packages/theming/src/utils/getColor.ts

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { scale, valid } from 'chroma-js';
+import { darken, lighten, rgba } from 'polished';
+import get from 'lodash.get';
+import memoize from 'lodash.memoize';
+import DEFAULT_THEME from '../elements/theme';
+import PALETTE from '../elements/palette';
+import { Hue, IGardenTheme } from '../types';
+
+const PALETTE_SIZE = Object.keys(PALETTE.blue).length;
+
+const adjust = (color: string, expected: number, actual: number) => {
+  if (expected !== actual) {
+    // Adjust darkness/lightness if color is not the expected shade
+    const amount = (Math.abs(expected - actual) / 100) * 0.05;
+
+    return expected > actual ? darken(amount, color) : lighten(amount, color);
+  }
+
+  return color;
+};
+
+/* convert the optional shade + offset to a shade for the given scheme */
+const toShade = (shade?: number | string, offset?: number, scheme?: 'dark' | 'light') => {
+  let _shade;
+
+  if (shade === undefined) {
+    _shade = scheme === 'dark' ? 500 : 700;
+  } else {
+    _shade = parseInt(shade.toString(), 10);
+
+    if (isNaN(_shade)) {
+      throw new TypeError(`Error: unexpected '${typeof shade}' type for color shade "${shade}"`);
+    }
+  }
+
+  return _shade + (offset || 0);
+};
+
+/* convert the given hue object to a hex color */
+const toHex = (
+  hue: Record<string | number, string>,
+  shade?: number | string,
+  offset?: number,
+  scheme?: 'dark' | 'light'
+) => {
+  const _shade = toShade(shade, offset, scheme);
+  let retVal = hue[_shade];
+
+  if (!retVal) {
+    const closestShade = Object.keys(hue)
+      .map(hueShade => parseInt(hueShade, 10))
+      .reduce((previous, current) => {
+        // Find the closest available shade within the given hue
+        return Math.abs(current - _shade) < Math.abs(previous - _shade) ? current : previous;
+      });
+
+    retVal = adjust(hue[closestShade], _shade, closestShade);
+  }
+
+  return retVal;
+};
+
+/* convert the given hue + shade to a color */
+const toColor = (
+  colors: Omit<IGardenTheme['colors'], 'base' | 'variables'>,
+  palette: IGardenTheme['palette'],
+  scheme: 'dark' | 'light',
+  hue: string,
+  shade?: number | string,
+  offset?: number,
+  transparency?: number
+) => {
+  let retVal;
+  let _hue: Hue =
+    colors[hue as keyof typeof colors] /* ex. `hue` = 'primaryHue' */ ||
+    hue; /* ex. `hue` = '#fd5a1e' */
+
+  if (Object.hasOwn(palette, _hue)) {
+    _hue = palette[_hue]; /* ex. `hue` = 'grey' */
+  }
+
+  if (typeof _hue === 'object') {
+    retVal = toHex(_hue, shade, offset, scheme);
+  } else if (valid(_hue)) {
+    if (shade === undefined) {
+      retVal = _hue;
+    } else {
+      const _colors = scale([PALETTE.white, _hue, PALETTE.black])
```

Comment:
Chaining `.mode('lab')` could result in more vibrant colors in some cases. 

Here's a [sandbox](https://stackblitz.com/edit/stackblitz-starters-sgcjtk?file=src%2FApp.tsx) highlighting the differences.

Reply:
Awesome. Similar to the previous comment, I'd like to continue forward and get some of this comparative analysis into our designer's hands for evaluation. To my eyes, the LAB side has a tendency to yellow/brown-out the colors. We know the designers are getting good leverage out of [Leonardo](https://github.com/adobe/leonardo/blob/main/packages/contrast-colors/lib/color.js), which defaults to RGB and then adds some [OKLCH sugar](https://github.com/adobe/leonardo/blob/ac2237b82cae7e10e5248d104c697ac70c168d14/packages/contrast-colors/lib/color.js#L120-L129) on top of `chroma.js`. But we also know that standard CSS advancements are happening fast, and Leonardo only [just graduated](https://www.npmjs.com/package/@adobe/leonardo-contrast-colors?activeTab=versions) from alpha after having been [dormant for nearly 2 years](https://github.com/adobe/leonardo/blob/main/packages/contrast-colors/CHANGELOG.md). Anyway, more to research in this space.

Reply:
> To my eyes, the LAB side has a tendency to yellow/brown-out the colors.

Good observation! 👁️ 

> defaults to RGB and then adds some [OKLCH sugar](https://github.com/adobe/leonardo/blob/ac2237b82cae7e10e5248d104c697ac70c168d14/packages/contrast-colors/lib/color.js#L120-L129) on top of chroma.js

Very interesting. Thanks for sharing this.


I agree. We can iteratively fine-tune this as we see fit. Great work!